File: 2019/1/1.js
let input = require('fs').readFileSync('input', 'utf8').split('\n').filter(x => x).map(x => parseInt(x))

let f = n => Math.max(0, Math.floor(n / 3) - 2)
console.log(input.map(f).reduce((a,b) => a + b))

let g = n => f(n) + (f(n) && g(f(n)))
console.log(input.map(g).reduce((a,b) => a + b))

File: 2019/1/1.ex
defmodule Main do
  @input File.stream!("input") |> Stream.map(&String.trim/1) |> Stream.map(&String.to_integer/1) |> Enum.to_list()

  defp f(n), do: max(0, floor(n / 3) - 2)

  defp g(n) when n <= 0, do: 0
  defp g(n), do: f(n) + g(f(n))

  def main() do
    {
      @input |> Enum.map(&f/1) |> Enum.sum(),
      @input |> Enum.map(&g/1) |> Enum.sum()
    }
  end
end
